### DISCUSSION TITLE
Some ideas

### DISCUSSION THREAD

**Discussion by szyper-sakrona** [Jakeler/ble-serial#41] [Ideas]
Hello!
Dear Mr. Jakeler - you had rescued my educational classes with children - your scripts works wonderfully with my BT makeblock Mbots. Thanks!
The only thing which could be corrected in your 'readme' was that the setup (on Windows) with com0com software should be more detailed - there is no COM9 created after "ble-com-setup.exe" execution. Instead there were 2 normal pairs of COM7-8 which is not needed. Please add one tip - the configuration of com0com (ex. using "setupg") should be set up that COM9 use Ports class (checkmark on) and BLE port is not using Ports class (no checkmark). Then the port is magically added and fun begins.

I have some ideas - instead of using cmdline I have to build simple GUI (I think about PySimpleGUI) that execute BLE-scan - and display the MAC id (also names of my mbots) to connect to - every time it could be different mbot so the idea is to choose the right robot.
Could you give any tips of executing scripts inside PySimpleGUI - using exe or .py? I do not have any experience in this (yet!).
Thanks and once again - you are lifesaver! Instead of buying original Bluetooth dongles (unnecessary, because my laptops have BT already builtin) we can use your software to control robots.

**Comment by Jakeler** [comment]
Hi @szyper-sakrona,
thanks for the feedback, I am really glad to hear that and awesome that you teach kids such topics!

About the setup tool: I am not sure why this happens, you should not have to change anything in the config manually. I just have it tested again on a clean Windows 10 virtual machine and the `COM9` got created as expected, otherwise you might have to reboot or re-run the setup script. If it reproducibly has this behavior then please create an issue and fill in your system info, so I can either adjust the readme or the code.

About your GUI idea: Scanning is pretty easy, I would recommend to interface directly with the `bleak` backend in python, instead of parsing the output of `ble-scan`. You can get some inspiration on the [scan script](https://github.com/Jakeler/ble-serial/blob/v2.4.1/ble_serial/scan/__init__.py), basically you can copy the `general_scan()` function and just replace the print loop [here](https://github.com/Jakeler/ble-serial/blob/v2.4.1/ble_serial/scan/__init__.py#L20) with a function that adds a result to the GUI.
Then for starting ble-serial with a result from your GUI you can use python [subproccess.run](https://docs.python.org/3/library/subprocess.html) to execute the exe with parameters, this is probably the most reliable and easy solution. I am doing something similar in the tests [here](https://github.com/Jakeler/ble-serial/blob/v2.4.1/tests/serial_handler.py#L49).
Let me know how goes! Currently I don't have much time, but I could make a minimal GUI if you have difficulties at the beginning.

**Reply by szyper-sakrona** [reply]
Hello, I managed to create simple GUI - but the first error is that your library is not wokring on my old Windows 7 laptop's. Please give me any hope that "bleak" could be run on my old machines...It is surprise...maybe any workarounds?
Thanks

**Reply by szyper-sakrona** [reply]
Uhm, I review current options and decided to upgrade to windows 10.
Anyway I'm currently introducing to Python and after first alpha version of my search and connect GUI - I'd want to improve the final executable.
Thanks for any support tips and your shared knowledge! Regards
